### ISSUE TITLE
diffsims is not giving this user the diffraction pattern they were expecting (from MMSE).

### ISSUE THREAD

**Issue by ndattani** [pyxem/diffsims#176]
Hello! One of our users at MMSE asked this question related to diffsims: https://mattermodeling.stackexchange.com/q/6096/5. Does anyone know how to answer it? It would really be appreciated, either if you could say why diffsims is not giving the user why they expected, or to recommend some other software for what they're trying to do, or to say why they can't do it in diffsims.

**Comment by din14970** [comment]
The reason it doesn't work is that the `calibration` parameter of the diffraction pattern is not set. A diffraction pattern in diffsims is represented by a list of `(x, y)` coordinates in projected reciprocal space, and an associated intensities `I`. Together they represent diffraction spots, and the main purpose of diffsims is to calculate `I` in these patterns depending on the microscope conditions (like sample orientation, acceleration voltage, ...) and create libraries of these simulated patterns for template matching (comparing to experimental data). A diffraction pattern can exist "etherially" in reciprocal space coordinates, but to convert them to an image one needs to know the size of a pixel in reciprocal space units.  By default, this value is 1, see https://github.com/pyxem/diffsims/blob/3c2540e520f7c8a6a60d24ec9b88844c4d2c8f83/diffsims/sims/diffraction_simulation.py#L50 meaning that 1 pixel will correspond to 1 angstrom^-1. Your user is thus creating a diffraction pattern image where all the diffraction spots she expects are all blurred together in the center.  In a sense, she just needs to "magnify" the diffraction pattern by setting a different calibration. Before the line

```python
dp= diffraction_pattern.get_diffraction_pattern(size=512, sigma=2)
```

the user should add a line

```python
diffraction_pattern.calibration = #some other value representing the number of angstrom^-1 / pixel
```

Hope this helps.

Also an important note for the user: since diffraction patterns in diffsims are essentially a structured point cloud, an unprocessed pattern is basically a zero image with some bright pixels. When we collect real diffraction patterns in the microscope, spots have a finite size due to imperfect optics, thermal vibrations, finite sample and beam size, dynamic diffraction,... All these effects are not considered in diffsims, it only uses the kinematical approximation. To make the pattern appear more like experimental patterns, the bright pixels are simply blurred with a 2D gaussian. This is not representative of a physical process and just serves as a sanity check when judging the patterns.

This comment did make me realize that this default behavior is not really desireable, it would make more sense for calibration to be `None` and for the function that creates an image to not work unless the user explicitly sets a calibration. I will have a look.

**Comment by hakonanes** [comment]
> This comment did make me realize that this default behavior is not really desireable, it would make more sense for calibration to be `None` and for the function that creates an image to not work unless the user explicitly sets a calibration.

I agree. A default option should produce a reasonable result.

**Comment by pc494** [comment]
Hi, apologies for the late arrival. If the user could send the .cif file to the pyxem team email (pyxem **dot** team **at** gmail **dot** com) I would be keen to double check what's happening here.

Best wishes,
Phillip

**Comment by din14970** [comment]
I have implemented a proposed solution in #175

**Comment by pc494** [comment]
#175 fixed this and has been merged, closing
